### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by dittos**

### CODE REVIEW

File: src/ui/CommentThread.module.css [hunk 1]
```diff
@@ -64,4 +64,28 @@
 }
 .EditingActions button {
   margin-right: 8px;
-}
\ No newline at end of file
+}
+
```
Comment (dittos) [inline src/ui/CommentThread.module.css:68]: asdf

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=src%2Fui%2FCommentThread.module.css)_</sub>

File: package.json [hunk 2]
```diff
@@ -33,17 +34,19 @@
     "redux": "^4.0.1",
     "redux-observable": "^1.0.0",
     "rxjs": "^6",
-    "typescript": "3.3.1"
+    "typescript": "^4.2.4"
   },
   "devDependencies": {
+    "apollo": "^2.33.2",
```
Comment (dittos) [inline package.json:40]: asdf

File: package.json [hunk 3]
```diff
@@ -3,6 +3,7 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@apollo/client": "^3.3.18",
```
Comment (dittos) [inline package.json:6]: afsddsaf

**Review by dittos — COMMENTED** [review]
dafsdsaf

**Review by dittos — COMMENTED** [review]
haha

**Review by dittos — COMMENTED** [review]
???

**Review by dittos — COMMENTED** [review]
???

File: package.json [hunk 4]
```diff
@@ -3,6 +3,7 @@
   "version": "0.1.0",
```
Comment (dittos) [inline package.json:3]: ffff

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=package.json)_</sub>

Comment (dittos) [inline package.json:3]: ggg

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=package.json)_</sub>

File: package.json (hunk 3 above)
Comment (dittos) [inline package.json:6]: added

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=package.json)_</sub>

File: package.json [hunk 5]
```diff
@@ -33,17 +34,19 @@
     "redux": "^4.0.1",
     "redux-observable": "^1.0.0",
     "rxjs": "^6",
-    "typescript": "3.3.1"
```
Comment (dittos) [inline package.json:36]: rem

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=package.json)_</sub>

File: package.json (hunk 4 above)
Comment (dittos) [inline package.json:3]: hhhh

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=package.json)_</sub>

Comment (dittos) [inline package.json:3]: zzz

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=package.json)_</sub>

File: package.json [hunk 6]
```diff
@@ -3,6 +3,7 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@apollo/client": "^3.3.18",
     "@blueprintjs/core": "^3.28.2",
     "@types/classnames": "^2.2.7",
     "@types/diff2html": "^0.0.4",
```
Comment (dittos) [inline package.json:9]: asdgsagds

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=package.json)_</sub>

File: package.json (hunk 4 above)
Comment (dittos) [reply package.json:3]: ggg

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=package.json)_</sub>

Comment (dittos) [inline package.json:3]: aaa

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=package.json)_</sub>

Comment (dittos) [inline package.json:3]: aaa

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=package.json)_</sub>

Comment (dittos) [inline package.json:3]: aaa

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=package.json)_</sub>

Comment (dittos) [reply package.json:3]: bbb

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=package.json)_</sub>

File: package.json [hunk 7]
```diff
@@ -3,6 +3,7 @@
   "version": "0.1.0",
   "private": true,
```
Comment (dittos) [inline package.json:4]: fff

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=package.json)_</sub>

Comment (dittos) [reply package.json:4]: ggg

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=package.json)_</sub>

File: package.json (hunk 4 above)
Comment (dittos) [reply package.json:3]: hhh

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=package.json)_</sub>

Comment (dittos) [reply package.json:3]: zzz

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=package.json)_</sub>

File: package.json [hunk 8]
```diff
@@ -3,16 +3,17 @@
   "version": "0.1.0",
```
Comment (dittos) [inline package.json:3]: asdf

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=package.json)_</sub>

Comment (dittos) [reply package.json:3]: qwer

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=package.json)_</sub>

Comment (dittos) [inline package.json:3]: ggg

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=package.json)_</sub>

**Review by dittos-test — APPROVED** [review]

File: src/stores/CommentStore.ts [hunk 9]
```diff
@@ -1,27 +1,34 @@
 import { combineEpics, ActionsObservable, StateObservable } from 'redux-observable';
-import { Subject, of } from 'rxjs';
```
Comment (dittos) [inline src/stores/CommentStore.ts:2]: asdf

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=src%2Fstores%2FCommentStore.ts)_</sub>

File: src/lib/Database.ts [hunk 10]
```diff
@@ -20,14 +20,14 @@ function refValues<T>(ref: firebase.database.Reference): Observable<T> {
   });
 }
 
-function reviewStatesRef(pullRequestId: number): firebase.database.Reference {
+function reviewStatesRef(pullRequestId: string): firebase.database.Reference {
   return firebase.database().ref(`reviewStates/${pullRequestId}/${getFirebaseUid()}`);
 }
 
-export function observeReviewStates(pullRequestId: number): Observable<{[fileId: string]: boolean}> {
+export function observeReviewStates(pullRequestId: string): Observable<{[fileId: string]: boolean}> {
```
Comment (dittos-test) [inline src/lib/Database.ts:27]: cmt

File: package.json [hunk 11]
```diff
@@ -23,7 +23,6 @@
     "fuzzaldrin-plus": "^0.6.0",
```
Comment (dittos-test) [inline package.json:23]: asfd

**Review by dittos-test — CHANGES_REQUESTED** [review]

**Review by dittos-test — COMMENTED** [review]
ggg

**Review by dittos-test — DISMISSED** [review]

File: src/lib/Database.ts (hunk 10 above)
Comment (dittos-test) [reply src/lib/Database.ts:27]: asfdf

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=src%2Flib%2FDatabase.ts)_</sub>

Comment (dittos-test) [reply src/lib/Database.ts:27]: hi

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=src%2Flib%2FDatabase.ts)_</sub>

File: package.json (hunk 11 above)
Comment (dittos-test) [inline package.json:23]: aaaa

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=package.json)_</sub>

Comment (dittos-test) [reply package.json:23]: bbbb

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=package.json)_</sub>

Comment (dittos-test) [reply package.json:23]: zzzz

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=package.json)_</sub>

File: package.json [hunk 12]
```diff
@@ -23,7 +23,6 @@
...
     "highlight.js": "^9.12.0",
     "http-link-header": "^0.8.0",
-    "marked": "^0.3.18",
```
Comment (dittos-test) [inline package.json:26]: hhhh

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=package.json)_</sub>

**Review by dittos-test — DISMISSED** [review]

File: package.json (hunk 11 above)
[the remark this answers is not in the document]
Comment (dittos) [reply package.json:23]: gg

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/diffmonster/pull/67?path=package.json)_</sub>

**Pull request merged**